lib/ronin/cli/commands/compress.rb

```
@@ -0,0 +1,163 @@
+                        @compression_method = :zlib_deflate
+                      end
+
+        option :name, short: '-n',
+                      value: {
+                        type: String
+                      },
+                      desc: 'compressed file name' do |name|
+                        @compressed_file_name = name
+                      end
+
+        description 'Compress the data'
+
+        man_page 'ronin-compress.1'
+
+        #
+        # The compression format.
+        #
+        # @return [:gzip, :zlib_deflate]
+        #
+        attr_reader :compression_method
+
+        #
+        # Initializes the `ronin compress` command.
+        #
+        # @param [Hash{Symbol => Object}] kwargs
+        #   Additional keyword arguments.
+        #
+        def initialize(**kwargs)
+          super(**kwargs)
+
+          @compression_method = :gzip
+        end
+
+        #
+        # Runs the `compress` sub-command.
+        #
+        # @param [Array<String>] files
+        #   File arguments.
+        #
+        def run(*files)
+          if files.empty?
+            super(*files)
+          else
+            raise "Files can be compressed using gzip only" if @compression_method != :gzip
+
+            Ronin::Support::Compression.gzip(compressed_file_name) do |gz|
+              files.each do |file|
+                File.open(file, 'rb') do |f|
+                  while (chunk = f.readpartial(4096))
+                    gz.write chunk
+                  end
+                end
+              rescue EOFError
+              rescue IOError => e
+                puts "Error reading file: #{e.message}"
+              end
+            end
+          end
+        end
+
+        #
+        # Reads and processes file.
+        #
+        # @param [String] path
+        #   The path to the file.
+        #
+        def process_file(file)
+          raise "Files can be compressed using gzip only" if @compression_method != :gzip
```

Should we raise an error, or just print some message and return?